sky/shell/ui/engine.cc

```diff
@@ -68,6 +69,32 @@ void Engine::BeginFrame(ftl::TimePoint frame_time) {
     sky_view_->BeginFrame(frame_time);
 }
 
+void Engine::RunFromSource(const std::string& main,
+                           const std::string& packages,
+                           const std::string& assets_directory) {
+  TRACE_EVENT0("flutter", "Engine::RunFromSource");
+  // Assets.
+  base::FilePath assets_directory_path = base::FilePath(assets_directory);
+  ConfigureDirectoryAssetBundle(assets_directory_path);
+  // .packages.
+  base::FilePath packages_path = base::FilePath(std::string(packages));
+  if (packages_path.empty()) {
+    base::FilePath main_dir = base::FilePath(main).DirName();
+    packages_path = main_dir.Append(FILE_PATH_LITERAL(".packages"));
+    if (!base::PathExists(packages_path)) {
+      packages_path = main_dir.Append(base::FilePath::kParentDirectory)
+                          .Append(FILE_PATH_LITERAL(".packages"));
+      if (!base::PathExists(packages_path))
+        packages_path = base::FilePath();
+    }
+  }
+  DartLibraryProviderFiles* provider = new DartLibraryProviderFiles();
+  dart_library_provider_.reset(provider);
```

Comment:
Actually undone because the call to LoadPackagesMap needs a DartLibraryProviderFiles